app/frontend/src/features/messages/groupchats/MembersDialog.tsx

```diff
@@ -0,0 +1,168 @@
+import Button from "../../../components/Button";
+import React, { useState } from "react";
+import {
+  Dialog,
+  DialogActions,
+  DialogTitle,
+  DialogContent,
+} from "../../../components/Dialog";
+import {
+  CircularProgress,
+  DialogProps,
+  List,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  makeStyles,
+} from "@material-ui/core";
+import { GroupChat } from "../../../pb/conversations_pb";
+import useUsers from "../../userQueries/useUsers";
+import Avatar from "../../../components/Avatar";
+import { User } from "../../../pb/api_pb";
+import useAuthStore from "../../auth/useAuthStore";
+import TextBody from "../../../components/TextBody";
+import { useMutation, useQueryClient } from "react-query";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { Error as GrpcError } from "grpc-web";
+import { service } from "../../../service";
+import Alert from "../../../components/Alert";
+
+const useStyles = makeStyles((theme) => ({
+  memberListItemTextContainer: {
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+  },
+  avatar: {
+    width: 30,
+    height: 30,
+  },
+  avatarContainer: {
+    minWidth: `calc(30px + ${theme.spacing(1)})`,
+  },
+}));
+
+function MemberListItem({
+  groupChatId,
+  member,
+  memberIsAdmin,
+  userIsAdmin,
+  setError,
+}: {
+  groupChatId: number;
+  member: User.AsObject;
+  memberIsAdmin: boolean;
+  userIsAdmin: boolean;
+  setError: (value: string) => void;
+}) {
+  const classes = useStyles();
+
+  const queryClient = useQueryClient();
+  const clearError = () => setError("");
+  const handleError = (error: GrpcError) => setError(error.message);
+  const invalidate = () => {
+    queryClient.invalidateQueries(["groupChatMessages", groupChatId]);
+    queryClient.invalidateQueries(["groupChats"]);
+    queryClient.invalidateQueries(["groupChat", groupChatId]);
+  };
+
+  const makeAdmin = useMutation<Empty, GrpcError, void>(
+    () => service.conversations.makeGroupChatAdmin(groupChatId, member),
+    {
+      onMutate: clearError,
+      onSuccess: invalidate,
+      onError: handleError,
+    }
+  );
+  const removeAdmin = useMutation<Empty, GrpcError, void>(
+    () => service.conversations.removeGroupChatAdmin(groupChatId, member),
+    {
+      onMutate: clearError,
+      onSuccess: invalidate,
+      onError: handleError,
+    }
+  );
+
+  const handleMakeAdmin = () => makeAdmin.mutate();
+  const handleRemoveAdmin = () => removeAdmin.mutate();
+
+  return (
+    <ListItem dense>
+      <ListItemAvatar className={classes.avatarContainer}>
+        <Avatar user={member} className={classes.avatar} />
+      </ListItemAvatar>
+      <ListItemText
+        disableTypography
+        className={classes.memberListItemTextContainer}
+      >
+        <TextBody>
+          {member.name}
+          {memberIsAdmin && " (admin)"}
+        </TextBody>
+        {userIsAdmin &&
+          (memberIsAdmin ? (
```

Comment:
These buttons don't line up properly on the smallest phone screen width (320px)
![Manage members dialog](https://user-images.githubusercontent.com/13669362/106396894-9f48e980-6402-11eb-9e77-64277df9114c.png)

But maybe it doesn't matter so much for now in this iteration? It looks like in the latest hi-fi design, we're shifting to using icons instead?

Reply:
That's because of the long name with no spaces to wrap. I added `noWrap` so names will be on one line now and truncated.